File: src/test/java/com/roscopeco/jasm/model/PrimArrayTests.java
package com.roscopeco.jasm.model;

public interface PrimArrayTests {
    void testBipushBastore(final byte[] in);
    byte testBaload(final byte[] in);
    void testCastore(final char[] in);
    char testCaload(final char[] in);
    void testDastore(final double[] in, final double testval);
    double testDaload(final double[] in);
    void testFastore(final float[] in, final float testval);
    float testFaload(final float[] in);
    void testIastore(final int[] in, final int testval);
    int testIaload(final int[] in);
    void testLastore(final long[] in, final long testval);
    long testLaload(final long[] in);

    int[] testNewarray(int size);
}

File: src/test/kotlin/com/roscopeco/jasm/model/PrimArrayTests.kt
package com.roscopeco.jasm.model

interface PrimArrayTests {
    fun testBipushBastore(input: ByteArray)
    fun testBaload(input: ByteArray): Byte
    fun testCastore(input: CharArray)
    fun testCaload(input: CharArray): Char
    fun testDastore(input: DoubleArray, testval: Double)
    fun testDaload(input: DoubleArray): Double
    fun testFastore(input: FloatArray, testval: Float)
    fun testFaload(input: FloatArray): Float
    fun testIastore(input: IntArray, testval: Int)
    fun testIaload(input: IntArray): Int
    fun testLastore(input: LongArray, testval: Long)
    fun testLaload(input: LongArray): Long
    fun testSastore(input: ShortArray, testval: Short)
    fun testSaload(input: ShortArray): Short
    fun testNewarray(size: Int): IntArray
}